flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -191,6 +194,11 @@ public void sendSplitRequest() {
 			public void sendSourceEventToCoordinator(SourceEvent event) {
 				operatorEventGateway.sendEventToCoordinator(new SourceEventWrapper(event));
 			}
+
+			@Override
+			public UserCodeClassLoader getUserCodeClassLoader() {
+				return (UserCodeClassLoader) getRuntimeContext().getUserCodeClassLoader();
```

Comment:
This looks like a cast between incompatible types.

Reply:
You can probably reuse some parts from `RuntimeContextInitializationContextAdapters` here.